pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java

```diff
@@ -311,4 +316,75 @@ public static void removeStarTrees(File indexDir)
     FileUtils.forceDelete(new File(segmentDirectory, StarTreeV2Constants.INDEX_FILE_NAME));
     FileUtils.forceDelete(new File(segmentDirectory, StarTreeV2Constants.INDEX_MAP_FILE_NAME));
   }
+
+  public static List<ExpressionContext> expressionContextFromFunctionParameters(
+      AggregationFunctionType aggregationFunctionType, Map<String, Object> functionParameters) {
+    List<ExpressionContext> expressionContexts = new ArrayList<>();
+
+    switch (aggregationFunctionType) {
+      case DISTINCTCOUNTHLL:
+      case DISTINCTCOUNTRAWHLL: {
+        if (functionParameters.containsKey(Constants.HLL_LOG2M_KEY)) {
+          expressionContexts.add(ExpressionContext.forLiteral(
```

Comment:
Ideally we want to construct a `LiteralContext` with proper type, e.g. `INT` for `log2m`